File: TimeCard/Extensions/CalendarExtension.swift
//
//  CalendarExtension.swift
//  TimeCard
//
//  Created by uhimania on 2025/10/02.
//

import Foundation

extension Calendar {
    func datesOf(year: Int, month: Int) -> [Date] {
        var dates: [Date] = []
        
        var date = Calendar.current.date(from: DateComponents(year: year, month: month))
        while date?.month == month {
            guard let d = date else { break }
            dates.append(d)
            date = Calendar.current.date(byAdding: DateComponents(day: 1), to: d)
        }
        
        return dates
    }
}

File: TimeCardClientAndroid/app/src/main/java/jp/uhimania/timecardclientandroid/data/CalendarExtension.kt
package jp.uhimania.timecardclientandroid.data

import java.util.Calendar
import java.util.Date

fun Calendar.datesOf(year: Int, month: Int): List<Date> {
    val dates = mutableListOf<Date>()
    val calendar = Calendar.getInstance()
    calendar.clear()
    calendar.set(year, month - 1, 1)

    var date = calendar.time
    while (date.month() == month) {
        dates.add(date)
        calendar.add(Calendar.DAY_OF_MONTH, 1)
        date = calendar.time
    }

    return dates.toList()
}
